### PR TITLE
Replace distro type

### LINKED ISSUE #342: Remove strong distro type
Today there is a `distro.Type` that is not internally used within syft (but is within `grype`).  We should consider starting to trust raw os-release data and pass it through to the JSON presenters. We should also consider removing this `distro.Type` type from syft and move it to grype.

**Additional context**:
This is from a discussion from https://anchorecommunity.slack.com/archives/C019BUXV7R6/p1614950344012600 and https://github.com/anchore/syft/pull/341

### PR DESCRIPTION
**Pull request by wagoodman**
Removes the `distro.Distro` object in favor for a new `linux.Release` struct. The existing `distro` object takes part of the raw information that is found during analysis and attempts to identify the release relative to a hard-coded enumeration. The remaining raw information discovered is thrown away. There are problems with this:

1. Syft is meant to raise up as much raw information as possible and attempt not to synthesize data or conclude about interpretations of the data. The distro object is in conflict with this.
2. Throwing away the unused information from the underlying data source hinders downstream use of such data
3. Grype is really in need of this `distro.Type` enumeration, which leads to unnecessary updates of syft to improve grype matches (e.g. adding a new supported distro).

The new `linux.Release` struct raises up information found in [`/*/os-release`](https://www.freedesktop.org/software/systemd/man/os-release.html), which is extremely common across all distributions.
 
This makes a breaking change to the JSON schema, thus, it is being bumped in this PR from 2 to 3. The existing `.distro.idLike` field  is a string. The underlying field is a space-separated list of distribution names which the current linux release is most close to. This field is being transformed into a list to eliminate the need for downstream consumers to split the field on space before use.

Closes #342

### CODE REVIEW

File: internal/anchore/import_package_sbom.go [hunk 1]
```diff
@@ -8,31 +8,82 @@ import (
...
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/wagoodman/go-progress"
 )
 
 type packageSBOMImportAPI interface {
 	ImportImagePackages(context.Context, string, external.ImagePackageManifest) (external.ImageImportContentResponse, *http.Response, error)
 }
 
+// importSBOM mirrors all elements found on the syftjson model format object relative to the anchore engine import schema.
+type importSBOM struct {
```
Comment (spiffcs) [inline internal/anchore/import_package_sbom.go:24]: I didn't see any notes about this in the PR description. Is this type/method needed for internal anchore purposes for when we're changing the schema with breaking changes `2 -> 3`?

**Review by spiffcs — COMMENTED** [review]
LGTM - I'll wait like the notes said for the grype and grype-db branches so we can check it all together but no real notes on this one.

File: internal/anchore/import_package_sbom.go (hunk 1 above)
Comment (wagoodman) [reply internal/anchore/import_package_sbom.go:24]: great point, let me update the PR description to reflect this change and why

File: syft/pkg/dpkg_metadata.go [hunk 2]
```diff
@@ -35,15 +35,15 @@ type DpkgFileRecord struct {
...
-func (m DpkgMetadata) PackageURL(d *distro.Distro) string {
-	if d == nil {
+func (m DpkgMetadata) PackageURL(distro *linux.Release) string {
+	if distro == nil {
 		return ""
 	}
 	pURL := packageurl.NewPackageURL(
 		// TODO: replace with `packageurl.TypeDebian` upon merge of https://github.com/package-url/packageurl-go/pull/21
 		// TODO: or, since we're now using an Anchore fork of this module, we could do this sooner.
 		"deb",
```
[the remark this answers is not in the document]
Comment (wagoodman) [reply syft/pkg/dpkg_metadata.go:45]: an upcoming PR is about to change this, I think in the way that you're describing (derive the value directly from the release ID)

File: test/integration/distro_test.go [hunk 3]
```diff
@@ -3,20 +3,22 @@ package integration
...
 
-	expected, err := distro.NewDistro(distro.Busybox, "1.31.1", "")
-	if err != nil {
-		t.Fatalf("could not create distro: %+v", err)
-	}
-
-	for _, d := range deep.Equal(sbom.Artifacts.Distro, &expected) {
-		t.Errorf("found distro difference: %+v", d)
+	expected := &linux.Release{
+		PrettyName: "BusyBox v1.31.1",
```
[the remark this answers is not in the document]
Comment (wagoodman) [reply test/integration/distro_test.go:15]: ** shrug **

**Review by spiffcs — APPROVED** [review]

**Comment by wagoodman** [comment]
I'm going to hold off on merging this until the sunsetting of grype-db is completed (since this changes the types shared between grype and grype-db)

**Pull request merged**